### PR TITLE
refs #30189: banner messages support

### PR DESCRIPTION
**Pull request by fabienph**

### CODE REVIEW

File: tests/acl/test_mod_wrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+#include "gdi/screen_info.hpp"
+#include "utils/timebase.hpp"
+#include "utils/colors.hpp"
+#include "keyboard/keymap2.hpp"
+#include "core/client_info.hpp"
+#include "RAIL/client_execute.hpp"
+#include "core/RDP/capabilities/window.hpp"
+#include "configs/config.hpp"
+#include "utils/sugar/zstring_view.hpp"
+//#include "utils/png.hpp"
```
Comment (jonathanpoelen) [inline tests/acl/test_mod_wrapper.cpp]: useless

File: tests/acl/test_mod_wrapper.cpp [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    gdi::OsdMsgUrgency omu = gdi::OsdMsgUrgency::NORMAL;
+
+    mod_wrapper_.display_osd_message(osd_message, omu);
+
+
+    constexpr auto png_filename = "osd_message1.png"_zv;
+
+    //dump_png24(png_filename.c_str(), front_, true);
+
+    RED_CHECK_IMG(front_, str_concat(IMG_TEST_PATH, png_filename).c_str());
```
Comment (jonathanpoelen) [inline tests/acl/test_mod_wrapper.cpp]: `RED_CHECK_IMG` already doing `dump_png24`.

-> `RED_CHECK_IMG(front_, IMG_TEST_PATH "osd_message1.png);`

File: tests/acl/test_mod_wrapper.cpp [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        TimeBase time_base_;
+        ClientInfo client_info_;
+        FakeFront front_;
+        const BGRPalette& palette_;
+        Keymap2 keymap_;
+        const Font& font_;
+        gdi::GraphicApi& gd_;
+        ClientExecute rail_client_execute_;
+        Inifile ini_;
+        ModWrapper mod_wrapper_;
```
Comment (jonathanpoelen) [inline tests/acl/test_mod_wrapper.cpp]: Only `mod_wrapper` and `front` are used, others should be private.
It is not necessary to have `font`, `gd` and `palette` as a member.

File: tests/acl/test_mod_wrapper.cpp [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    };
+
+    using namespace std::string_view_literals;
+}
+
+RED_FIXTURE_TEST_CASE(TestOSDMessageDisplay_NormalUrgency,
+                      TestOSDMessageDisplayFixture)
+{
+    constexpr auto osd_message = "Hello World !"sv;
+    gdi::OsdMsgUrgency omu = gdi::OsdMsgUrgency::NORMAL;
```
Comment (jonathanpoelen) [inline tests/acl/test_mod_wrapper.cpp]: Why not `constexpr auto` as with `osd_message` ?

File: src/core/session.cpp [hunk 5]
```diff
@@ -594,6 +594,30 @@ class Session
         }
 
         mod_wrapper.set_mod(next_state, mod_pack);
+
+        if (this->ini.get<cfg::context::banner_activate>())
+        {
+            this->ini.set<cfg::context::banner_activate>(false);
```
Comment (jonathanpoelen) [inline src/core/session.cpp]: based on the content of `cfg::context::banner_message`, `banner_activate` is not needed anymore.

File: src/gdi/osd_api.hpp [hunk 6]
```diff
@@ -27,16 +27,26 @@ Author(s): Proxy Team
...
+        virtual void display_osd_message(std::string_view message,
+                                         OsdMsgUrgency omu = OsdMsgUrgency::NORMAL) = 0;
         virtual ~OsdApi() = default;
     };
 
     struct NullOsd : gdi::OsdApi
     {
-        void display_osd_message(std::string_view message) override
+        void display_osd_message(std::string_view message,
+                                 OsdMsgUrgency) override
```
Comment (jonathanpoelen) [inline src/gdi/osd_api.hpp]: There should be a default parameter, as for the base class.

File: tools/sesman/sesmanworker/sesman.py [hunk 7]
```diff
@@ -2813,6 +2815,33 @@ def _manage_pm(self):
...
+        if banner_activate is False or not banner_message:
+            data_to_send = {u"banner_activate" : False}
+        else:
+            if (banner_type == "info"
+                or (banner_type != "warn" and banner_type != "alert")):
+                banner_type = 0
+            elif banner_type == "warn":
+                banner_type = 1
+            elif banner_type == "alert":
+                banner_type = 2
```
Comment (jonathanpoelen) [inline tools/sesman/sesmanworker/sesman.py]:
```py
                  if banner_type == "alert":
                      banner_type = 2
                  elif banner_type == "warn":
                      banner_type = 1
                  else # banner_type == "info":
                      banner_type = 0
```

**Review by jonathanpoelen — CHANGES_REQUESTED** [review]

File: tests/acl/test_mod_wrapper.cpp (hunk 4 above)
Comment (fabienph) [reply tests/acl/test_mod_wrapper.cpp]: It's already `constexpr auto` the `osd_message` variable (to force the instruction execution at compile time)

**Review by jonathanpoelen — APPROVED** [review]

**Pull request merged**